students/BrianNisonger/session9/mailroom.py

```diff
@@ -0,0 +1,197 @@
+
+    def make_filename(self):
+        """Turns a donor name into a filename"""
+        return f'{self.name.replace(" ","_")}.txt'
+
+    def make_report_string(self, value):
+        """Makes a single line of the report(Name,Total Donations,Num of Donatioms, Average Donations)"""
+        return f'{self.name:<20} ${self.make_total(value):<15.2f} {self.num_donations(value):<15} ${self.make_average(value):<15.2f}'
+
+    def __lt__(self, other):
+        """less than"""
+        return (self.make_total(self.donation) > other.make_total(
+            other.donation))
+
+    def __gt__(self, other):
+        """greater than"""
+        return (self.make_total(self.donation) < other.make_total(
+            other.donation))
+
+    def __eq__(self, other):
+        """equal"""
+        return (self.make_total(self.donation) == other.make_total(
+            other.donation))
+
+    def __str__(self):
+        """Returns name of the donor if str is called"""
+        return self.name
+
+    def sort_key(self):
+        """What to sort on"""
+        return (self.make_total(self.donation))
+
+
+class Donors():
+    """ This contains a list of classes, used to store multiple instances of the Donor class for grouping"""
+
+    def __init__(self, donor_list):
+        self.donor_list = donor_list
+
+    def append(self, cls):
+        """Adds another donor to the donor_list"""
+        self.donor_list.append(cls)
+
+    def list_donors(self):
+        """returns a list containing the names of all the donors in the donor list"""
+        return [donor.name for donor in self.donor_list]
+
+    def make_headers_string(self):
+        """Header for the report"""
+        headers = ("Donor Names", "Total Given", "Num Gifts", "Average Gifts")
+        header_string = f'{headers[0]:<20} {headers[1]:<15} {headers[2]:<15} {headers[3]:<15}'
+        return header_string
+
+    def sort_donors(self):
+        """Takes a list of donors and sorts by total donations"""
+        self.donor_list = sorted(self.donor_list)
+
+    def make_report(self):
+        """Combines the report header with individual report strings"""
+        report = self.make_headers_string() + "\n"
+        self.sort_donors()
+        for donor in self.donor_list:
+            report += donor.make_report_string(donor.donation) + "\n"
+        return report
+
+    def make_thank_notes(self):
+        """Provides a list containing all the thank you notes for Donors, Not Implemented in final code"""
+        self.thank_notes = []
+        for donor in self.donor_list:
+            self.thank_notes.append(donor.make_thank_you())
+        return self.thank_notes
+
+    def find_donor(self, value):
+        """Takes a string value and returns a donor class if donor name matches string, or a empty list if not"""
+        donor_class = []
+        for donor in self.donor_list:
```

Comment:
maybe you want to use a dict for faster lookup, but this is fine too.

It'll get slow if the list gets big -- but then you can refactor -- that's one good reason to put this in a class!